syntax = "proto3";
//import "google/protobuf/timestamp.proto";

package Core;

message TaskID {
  int32 value = 1;
}

message Task {
  string title = 1;
  enum Priority {
    NONE = 0;
    LOW = 1;
    MEDIUM = 2;
    HIGH = 3;
  }
  Priority priority = 2;
//  google.protobuf.Timestamp due_date = 3;
  int64  due_date = 3;
  repeated string labels = 4;
  bool is_complete = 5;
}

message TaskEntity {
  TaskID id = 1;
  Task data = 2;
  TaskID parent = 3;
}
